app.py

```diff
@@ -0,0 +1,151 @@
+        quotes=list(mongo.db.quotes.find()),
+        sources=list(mongo.db.sources.find())
+        )
+
+
+
+@app.route('/get_language/<language_id>') 
+def get_language(language_id):
+    """
+    This function will load the getlanguage.html that displays the quotes list depending on the chosen language
+    """
+    return render_template('getlanguage.html',
+        language=mongo.db.languages.find_one({'_id': ObjectId(language_id)}),
+        quotes=list(mongo.db.quotes.find()),
+        languages=list(mongo.db.languages.find())
+        )
+
+@app.route('/get_source/<source_id>')
+def get_source(source_id):
+    """
+    This function will load the getsource.html that displays the quotes list depending on the chosen source
+    """
+    return render_template('getsource.html',
+        source=mongo.db.sources.find_one({'_id': ObjectId(source_id)}),
+        quotes=list(mongo.db.quotes.find()),
+        sources=list(mongo.db.sources.find())
+        )
+
+
+@app.route('/quote/add', methods=['GET','POST'])
+def create_quote():
+    """
+    This function will allow the user to add a new quote
+    """
+    if request.method == 'POST':
+        quote = mongo.db.quotes
+        quote.insert_one(request.form.to_dict())
+        return redirect(url_for('home'))
+
+    return render_template('createquote.html',
+        quotes=list(mongo.db.quotes.find()),
+        category=list(mongo.db.categories.find()),
+        sources=list(mongo.db.sources.find()),
+        languages=list(mongo.db.languages.find())
+        )
+
+
+@app.route('/edit/<quote_id>', methods=['GET','POST'])
+def modify(quote_id):
+    """
+    This function will allow the user to edit an existing quote
+    """
+    if request.method == 'POST':
+        quote = mongo.db.quotes
+        quote.update({'_id': ObjectId(quote_id)}, {
+            'quote_category': request.form.get('quote_category'),
+            'quote_text': request.form.get('quote_text'),
+            'quote_author': request.form.get('quote_author'),
+            'quote_source': request.form.get('quote_source'),
+            'quote_source_name': request.form.get('quote_source_name'),
+            'quote_language': request.form.get('quote_language')
+        })
+        return redirect(url_for('home'))  # cannot redirect to a differen
+
+    return render_template('modifyquote.html',
+        quote=mongo.db.quotes.find_one({'_id': ObjectId(quote_id)}),
+        category=list(mongo.db.categories.find()),
+        sources=list(mongo.db.sources.find()),
+        languages=list(mongo.db.languages.find()))
+
+
+
+@app.route('/delete_quote/<quote_id>')
+def delete_quote(quote_id):
+    """
+    This function will allow the user to edit an existing quote
+    """
+    quote=mongo.db.quotes
+    quote.delete_one({'_id': ObjectId(quote_id)})
+    flash("Quote deleted")
+    return redirect(url_for('home'))
+
+
+if __name__ == '__main__':
+    app.run(host=os.getenv('IP'), port=os.getenv('PORT'), debug=True)
```

Comment:
debug should be false in production

Reply:
Done. I set the DEBUG variable = true in the .env file and equal to false in the heroku variables settings